File: backend/app/schemas/mentor.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any


class ChatMessage(BaseModel):
    role: str          # "user" or "model"
    text: str


class MentorChatRequest(BaseModel):
    message: str
    history: Optional[List[ChatMessage]] = []
    user_profile: Optional[Dict[str, Any]] = None


class MentorChatResponse(BaseModel):
    reply: str

File: frontend/src/api/mentor.ts
import { apiClient } from './client';

export interface ChatMessage {
  role: 'user' | 'model';
  text: string;
}

export interface MentorChatResponse {
  reply: string;
}

export const mentorApi = {
  chat: (message: string, history: ChatMessage[] = [], user_profile?: Record<string, any>) =>
    apiClient<MentorChatResponse>('/mentor/chat', {
      method: 'POST',
      data: { message, history, user_profile },
    }),
};
